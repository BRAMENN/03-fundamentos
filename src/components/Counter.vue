<template>
  <h2>{{ customTitle }}</h2>
  <p>
    {{ counter }} <sup>{{ pow }}</sup> = {{ squareCounter }}
  </p>

  <div>
    <button @click="decrease()">-1</button>
    <button @click="increase()">+1</button>
  </div>
</template>

<script>
export default {
  props: {
    title: String,
    start: {
      type: Number,
      default: 5,
      //   required: true,
      validator(value) {
        return value >= 0;
      },
    },
  },

  data() {
    return {
      counter: this.start,
      pow: 3,
    };
  },

  methods: {
    increase() {
      this.counter += 1;
    },

    decrease() {
      this.counter -= 1;
    },
  },

  computed: {
    squareCounter() {
      return Math.pow(this.counter, this.pow);
    },
    customTitle() {
      return this.title ? this.title : "Counter";
    },
  },
};
</script>

<style>
button {
  background-color: black;
  border-radius: 5px;
  color: aliceblue;
  margin: 0 5px;
  cursor: pointer;
  padding: 5px 15px;
  transition: 0.3s ease-in-out;
}

button:hover {
  background-color: rgba(0, 0, 0, 0.466);
  transition: 0.3s ease-in-out;
}
</style>
