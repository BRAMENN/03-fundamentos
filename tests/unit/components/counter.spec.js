import { shallowMount } from '@vue/test-utils'
import Counter from '@/components/Counter'

//Prueba que evalua si un componente cambio
describe('Counter Component', () => {
    test('Debe hacer match con el snapshot', () => {
        let wrapper = shallowMount(Counter)
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('h2 debe de tener el valor por defecto', () => {
        let wrapper = shallowMount(Counter)
        let h2Value = wrapper.find("h2").text()
        expect(h2Value).toBe("Counter")
    })
})