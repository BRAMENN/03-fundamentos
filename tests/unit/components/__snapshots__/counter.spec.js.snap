// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe hacer match con el snapshot 1`] = `
<h2>Counter</h2>
<p>5 <sup>3</sup> = 125</p>
<div><button>-1</button><button>+1</button></div>
`;
